modules/javafx.graphics/src/main/java/com/sun/glass/ui/HeaderButtonOverlay.java

```diff
@@ -0,0 +1,709 @@
+                     BASELINE_OFFSET_SAME_AS_HEIGHT, Insets.EMPTY, true, true,
+                     HPos.LEFT, VPos.TOP, false);
+
+        layoutInArea(button3, button3X, getButtonOffsetY(button3Height), button3Width, button3Height,
+                     BASELINE_OFFSET_SAME_AS_HEIGHT, Insets.EMPTY, true, true,
+                     HPos.LEFT, VPos.TOP, false);
+    }
+
+    @Override
+    public boolean usesMirroring() {
+        return false;
+    }
+
+    @Override
+    public List<CssMetaData<? extends Styleable, ?>> getCssMetaData() {
+        return METADATA;
+    }
+
+    private static double boundedWidth(Node node) {
+        return node.isManaged() ? boundedSize(node.minWidth(-1), node.prefWidth(-1), node.maxWidth(-1)) : 0;
+    }
+
+    private static double boundedHeight(Node node) {
+        return node.isManaged() ? boundedSize(node.minHeight(-1), node.prefHeight(-1), node.maxHeight(-1)) : 0;
+    }
+
+    private static double boundedSize(double min, double pref, double max) {
+        return Math.min(Math.max(pref, min), Math.max(min, max));
+    }
+
+    private class ButtonRegion extends Region {
+
+        private static final CssMetaData<ButtonRegion, Number> BUTTON_ORDER_METADATA =
+            new CssMetaData<>("-fx-button-order", StyleConverter.getSizeConverter()) {
+                @Override
+                public boolean isSettable(ButtonRegion node) {
+                    return true;
+                }
+
+                @Override
+                public StyleableProperty<Number> getStyleableProperty(ButtonRegion region) {
+                    return region.buttonOrder;
+                }
+            };
+
+        private static final List<CssMetaData<?, ?>> METADATA =
+            Stream.concat(getClassCssMetaData().stream(), Stream.of(BUTTON_ORDER_METADATA)).toList();
+
+        /**
+         * Specifies the layout order of this button relative to the other buttons.
+         * Buttons with a lower value are laid out before buttons with a higher value.
+         * <p>
+         * This property corresponds to the {@code -fx-button-order} CSS property.
+         */
+        private final StyleableIntegerProperty buttonOrder =
+            new SimpleStyleableIntegerProperty(BUTTON_ORDER_METADATA, this, "buttonOrder") {
+                @Override
+                protected void invalidated() {
+                    requestParentLayout();
+
+                    HeaderButtonOverlay.this.orderedButtons.sort(
+                        Comparator.comparing(ButtonRegion::getButtonOrder));
+                }
+            };
+
+        private final Region glyph = new Region();
+        private final HeaderButtonType type;
+
+        ButtonRegion(HeaderButtonType type, String styleClass, int order) {
+            this.type = type;
+            orderedButtons.add(this);
+            buttonOrder.set(order);
+            glyph.getStyleClass().setAll("glyph");
```

Comment:
"glyph": though apt, the name might be somewhat confusing.  maybe "icon"?

Reply:
I considered "icon", but still think that "glyph" captures the meaning a tiny bit better.